### PR TITLE
Add a multi timeseries return format

### PR DESCRIPTION
**Pull request by iwysiu**
Add support for returning data in the multi format for timeseries. The plan has shifted a bit: instead of return the multi format for timeseries instead of wide (what's described in the issue) I'm instead adding a multi return format for users who want a timeseries format for data with a lot of labels that don't have points for every time. This avoids creating a massive change for other users of the library.

Part of #96

### CODE REVIEW

File: query.go [hunk 1]
```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
...
 		return data.Frames{frame}, nil
-	}
-
-	if query.Format == FormatOptionTrace {
+	case FormatOptionTrace:
 		frame.Meta.PreferredVisualization = data.VisTypeTrace
 		return data.Frames{frame}, nil
-	}
+	case FormatOptionMulti:
+		if frame.TimeSeriesSchema().Type == data.TimeSeriesTypeLong {
```
Comment (njvrzm) [inline query.go:184]: I'm guessing the 3-plus-year-old deprecation warning on this method was... optimistic. (There are lots of other recent uses too.)

File: query.go [hunk 2]
```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
...
+			}
 
-	if err != nil {
-		return nil, err
+			frames, err := timeseries.LongToMulti(&timeseries.LongFrame{frame})
+			if err != nil {
+				return nil, err
+			}
+			return frames.Frames(), nil
+		}
```
Comment (njvrzm) [inline query.go:196]: Should it be an error to specifiy `FormatOptionMulti` with a different type here? (I don't expect so, mostly asking for my own edification)

File: query.go [hunk 3]
```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
 	frame.Meta.ExecutedQueryString = query.RawSQL
 	frame.Meta.PreferredVisualization = data.VisTypeGraph
 
-	if query.Format == FormatOptionTable {
-		frame.Meta.PreferredVisualization = data.VisTypeTable
-		return data.Frames{frame}, nil
+	count, err := frame.RowLen()
```
Comment (njvrzm) [inline query.go:165]: We could move this even higher, right, like right after the first error check?

File: query_test.go [hunk 4]
```diff
@@ -104,3 +106,29 @@ func TestQuery_Timeout(t *testing.T) {
...
+		require.NoError(t, err)
+
+		require.Equal(t, frame.Fields[0].Type(), data.FieldTypeTime)
+		require.Equal(t, frame.Fields[0].Len(), 2)
+		require.Equal(t, frame.Fields[0].At(0).(time.Time), time1)
+		require.Equal(t, frame.Fields[0].At(1).(time.Time), time2)
+
+		require.Equal(t, frame.Meta.Type, data.FrameTypeTimeSeriesLong)
+		require.Equal(t, frame.Meta.TypeVersion, data.FrameTypeVersion{0, 1})
+	})
```
Comment (njvrzm) [inline query_test.go:133]: Maybe add a test case for the error return on null values?

File: query.go [hunk 5]
```diff
@@ -158,40 +162,78 @@ func getFrames(rows *sql.Rows, limit int64, converters []sqlutil.Converter, fill
...
+	if err != nil {
+		return nil, err
+	}
+	if count == 0 {
+		return nil, ErrorNoResults
 	}
 
-	if query.Format == FormatOptionLogs {
+	switch query.Format {
```
Comment (njvrzm) [inline query.go:173]: This is entirely a matter of taste, and I'm not sure I actually prefer it, but I might structure this a bit differently to make the exceptional case stand out a little more: https://github.com/grafana/sqlds/compare/iwysiu/96...njvrzm/96-semi_suggestion (I can't add this as a suggestion since the range crosses deleted lines, sigh)

**Review by njvrzm — APPROVED** [review]
A few minor comments, but 👍🏻 as far as can see.

File: query.go (hunk 3 above)
Comment (iwysiu) [reply query.go:162]: done

File: query.go (hunk 5 above)
Comment (iwysiu) [reply query.go:173]: I think it makes sense. Done.

File: query.go (hunk 1 above)
Comment (iwysiu) [reply query.go:175]: Yeah, the dataplane repo has a new experimental implementation of timeseries that they're working on, but it's not there yet.

File: query.go (hunk 2 above)
Comment (iwysiu) [reply query.go:187]: No, I don't think it's an error. (Though we're not supporting wide to multi, I suspect that's not a use case we expect)

File: query_test.go (hunk 4 above)
Comment (iwysiu) [reply query_test.go:133]: done

**Review by iwysiu — COMMENTED** [review]
Thanks for the suggestions!

File: query.go [hunk 6]
```diff
@@ -18,14 +20,16 @@ import (
 type FormatQueryOption uint32
 
 const (
-	// FormatOptionTimeSeries formats the query results as a timeseries using "WideToLong"
+	// FormatOptionTimeSeries formats the query results as a timeseries using "LongToWide"
```
Comment (sarahzinger) [inline query.go:23]: so was this comment inaccurate before is this changing?

**Review by sarahzinger — COMMENTED** [review]
Ok 2 things:
- does something being a log, trace, or table affect it's long/wide/multi-ness? I am concerned we might be losing preferred visualizations potentially? It seems like were kind of combining concepts that maybe shouldn't be combined? Maybe frame.TimeSeriesSchema().Type should equal multi rather than adding this to the formatoptiontypes the way we have TimeSeriesTypeLong? Feel free to tell me I'm wrong about that haha. 
- I think we should make tests here for all the other option combos. Sorry I know it's annoying, but it seems really easy to accidentally make a breaking change when it's like an if case inside a switch case kind of thing.

**Review by sarahzinger — APPROVED** [review]
Talked about it in person, feeling good about it
